File: rules/python/weak-ciphers.py
"""Test fixtures for pqc-semgrep-rules / Python / classically-broken ciphers."""

from Crypto.Cipher import AES, ARC4, DES, DES3


def vulnerable():
    # ruleid: pqc.python.des
    DES.new(b"8bytekey", DES.MODE_ECB)

    # ruleid: pqc.python.3des
    DES3.new(b"24bytekeyforthis________", DES3.MODE_ECB)

    # ruleid: pqc.python.rc4
    ARC4.new(b"key")


def safe():
    # ok: pqc.python.des
    AES.new(b"32bytekeyforaesgcm______________", AES.MODE_GCM)

File: rules/go/weak-ciphers.go
// Test fixtures for pqc-semgrep-rules / Go / classically-broken ciphers.
package main

import (
	"crypto/aes"
	"crypto/des"
	"crypto/rc4"
)

func vulnerableCiphers() {
	// ruleid: pqc.go.des
	_, _ = des.NewCipher(make([]byte, 8))

	// ruleid: pqc.go.3des
	_, _ = des.NewTripleDESCipher(make([]byte, 24))

	// ruleid: pqc.go.rc4
	_, _ = rc4.NewCipher(make([]byte, 16))
}

func safeCiphers() {
	// ok: pqc.go.des
	_, _ = aes.NewCipher(make([]byte, 32))
}
